driven/storage/adapter.go

```diff
@@ -3476,86 +3471,127 @@ func (sa *Adapter) UpdateAuthTypes(ID string, code string, description string, i
+	filter := bson.M{"core_host": sa.host}
+	var result []model.ServiceRegistration
+	err := sa.db.serviceRegistrations.Find(filter, &result, nil)
 	if err != nil {
-		return nil, errors.WrapErrorAction(logutils.ActionFind, model.TypeServiceReg, &logutils.FieldArgs{"service_id": "all"}, err)
+		return nil, errors.WrapErrorAction(logutils.ActionFind, model.TypeServiceReg, &logutils.FieldArgs{"core_host": sa.host, "service_id": "all"}, err)
 	}
 
 	if result == nil {
-		result = []model.ServiceReg{}
+		result = []model.ServiceRegistration{}
 	}
 
 	return result, nil
 }
 
 // FindServiceRegs fetches the requested service registration records
-func (sa *Adapter) FindServiceRegs(serviceIDs []string) []model.ServiceReg {
+func (sa *Adapter) FindServiceRegs(serviceIDs []string) []model.ServiceRegistration {
 	return sa.getCachedServiceRegs(serviceIDs)
 }
 
 // FindServiceReg finds the service registration in storage
-func (sa *Adapter) FindServiceReg(serviceID string) (*model.ServiceReg, error) {
+func (sa *Adapter) FindServiceReg(serviceID string) (*model.ServiceRegistration, error) {
 	return sa.getCachedServiceReg(serviceID)
 }
 
 // InsertServiceReg inserts the service registration to storage
-func (sa *Adapter) InsertServiceReg(reg *model.ServiceReg) error {
-	_, err := sa.db.serviceRegs.InsertOne(reg)
+func (sa *Adapter) InsertServiceReg(reg *model.ServiceRegistration) error {
+	reg.CoreHost = sa.host
+	_, err := sa.db.serviceRegistrations.InsertOne(reg)
 	if err != nil {
-		return errors.WrapErrorAction(logutils.ActionInsert, model.TypeServiceReg, &logutils.FieldArgs{"service_id": reg.Registration.ServiceID}, err)
+		return errors.WrapErrorAction(logutils.ActionInsert, model.TypeServiceReg, &logutils.FieldArgs{"core_host": sa.host, "service_id": reg.Registration.ServiceID}, err)
 	}
 
 	return nil
 }
 
 // UpdateServiceReg updates the service registration in storage
-func (sa *Adapter) UpdateServiceReg(reg *model.ServiceReg) error {
-	filter := bson.M{"registration.service_id": reg.Registration.ServiceID}
-	err := sa.db.serviceRegs.ReplaceOne(filter, reg, nil)
+func (sa *Adapter) UpdateServiceReg(reg *model.ServiceRegistration) error {
+	reg.CoreHost = sa.host
+	filter := bson.M{"core_host": reg.CoreHost, "registration.service_id": reg.Registration.ServiceID}
+	err := sa.db.serviceRegistrations.ReplaceOne(filter, reg, nil)
 	if err != nil {
-		return errors.WrapErrorAction(logutils.ActionUpdate, model.TypeServiceReg, &logutils.FieldArgs{"service_id": reg.Registration.ServiceID}, err)
+		return errors.WrapErrorAction(logutils.ActionUpdate, model.TypeServiceReg, &logutils.FieldArgs{"core_host": reg.CoreHost, "service_id": reg.Registration.ServiceID}, err)
 	}
 
 	return nil
 }
 
 // SaveServiceReg saves the service registration to the storage
-func (sa *Adapter) SaveServiceReg(reg *model.ServiceReg, immediateCache bool) error {
+func (sa *Adapter) SaveServiceReg(reg *model.ServiceRegistration, immediateCache bool) error {
```

Comment:
I think we may be able to merge `SaveServiceReg` with `UpdateServiceReg` by adding an `upsert` argument.

Reply:
I think we can actually keep these separate to stick to our standard storage function definitions